src/services/iota.js

```diff
@@ -46,14 +59,83 @@ const queryTransactions = (iotaProvider, addresses) =>
     );
   });
 
-const checkUploadPercentage = addresses =>
+const skinnyQueryTransactions = (iotaProvider, addresses) =>
   new Promise((resolve, reject) => {
-    queryTransactions(IotaA, addresses).then(transactions => {
-      const percentage = transactions.length / addresses.length * 100;
-      resolve(percentage);
-    });
+    iotaProvider.api.findTransactions(
+      { addresses },
+      (error, transactionHashes) => {
+        if (error) {
+          console.log("IOTA ERROR: ", error);
+        }
+        resolve(transactionHashes);
+      }
+    );
   });
 
+const initializePolling = addresses =>
+  new Promise((resolve, reject) => {
```

Comment:
This isn't async, so doesn't need to be wrapped in a promise.

Reply:
Fixed